src/test/java/com/wb/WeatherForecastApiApplicationUnitTests.java

```diff
@@ -1,85 +1,54 @@
 package com.wb;
 
-import static org.junit.Assert.assertEquals;
-import static org.mockito.Mockito.when;
-
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
-
-import com.wb.controller.WeatherForecastController;
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.Mockito;
+import static org.springframework.test.web.servlet.ResultMatcher.matchAll;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
 
 import com.wb.payload.WeatherMapper;
 import com.wb.service.WeatherService;
 
-@RunWith(SpringRunner.class)
 @SpringBootTest
+@AutoConfigureMockMvc
 public class WeatherForecastApiApplicationUnitTests {
-    @Mock
-    private WeatherService weatherService;
-    @Mock
-    Future<WeatherMapper> futureWeather = new CompletableFuture<>();
-    @InjectMocks
-    WeatherForecastController controller = new WeatherForecastController(weatherService);
-    WeatherMapper weatherMapper;
 
-    @org.junit.jupiter.api.Test
-    public void contextLoads() {
-    }
+    private final WeatherService weatherService;
+    private final MockMvc mockMvc;
```

Comment:
`mockMvc` sluzy do mockowanie serwera Twojej aplikacji. Tak żebyś mógł przetestowac SWOJ kotnroller bez stawiania serwera. Po prostu trwa to szybciej a testuje mniej wiecej to samo.  Mam wrazenie ze probowales mockowac zewnetrzne API tym obiektem. To tak nie działa.